geth/whisper/mailservice.go

```diff
@@ -0,0 +1,46 @@
+package whisper
+
+import (
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/status-im/status-go/geth/common"
+)
+
+// MailService is a service that provides some additional Whisper API.
+type MailService struct {
+	nodeManager common.NodeManager
+}
+
+// NewMailService returns a new MailService.
+func NewMailService(nodeManager common.NodeManager) *MailService {
+	return &MailService{nodeManager}
+}
+
+// Protocols returns a new protocols list. In this case, there are none.
+func (s *MailService) Protocols() []p2p.Protocol {
+	return []p2p.Protocol{}
+}
+
+// APIs returns a list of new APIs.
+func (s *MailService) APIs() []rpc.API {
+	return []rpc.API{
+		{
+			Namespace: "shh",
+			Version:   "1.0",
+			Service:   NewMailServicePublicAPI(s.nodeManager),
+			Public:    true,
```

Comment:
Is it ok to have this method public? Public means it can be executed via HTTP. Private means it's available through inproc and IPC communication channels.

Reply:
It's ok for me.

Reply:
It's ok for me too